src/solvers/refinement/README.md

```diff
@@ -6,59 +6,101 @@
 
 \section string_solver_interface String solver interface
 
+The basic role of the solver is to answer whether the set of equations given
```

Comment:
Why does the string solver come first here? The directory contains various lazy/refinement-based solvers that  build upon the (eager) decision procedure bv_pointerst -- this should be explained first. The string solver is only one of these, enabling a theory combination of bv_pointerst with strings.

Reply:
I don't have a good knowledge of the other solvers, someone more knowledgeable about these should document them.

Reply:
@peterschrammel The task here was to document the string solver. Documentation of other parts would of course be useful, but it's out of the scope of this pr.